File: user-api/src/main/java/hu/aut/bme/userapi/feature/users/dto/UserMappers.java
package hu.aut.bme.userapi.feature.users.dto;

import hu.aut.bme.userapi.domain.User;

public class UserMappers {
    public static UserResponse userToUserResponse(User user) {
        return UserResponse.builder()
                .email(user.getEmail())
                .username(user.getUsername())
                .id(user.getId())
                .build();
    }

    public static User createUserRequestToUser(CreateUserRequest createUserRequest) {
        return User.builder()
                .username(createUserRequest.getUsername())
                .email(createUserRequest.getEmail())
                .build();
    }
}

File: blog-api/src/main/kotlin/hu/bme/aut/blogapi/feature/users/dto/UserMappers.kt
package hu.bme.aut.blogapi.feature.users.dto

import hu.bme.aut.blogapi.domain.User

fun User.toUserResponse(): UserResponse {
    return UserResponse(
            id = this.id!!,
            username = username,
            email = email
    )
}

fun CreateUserRequest.toUser(): User {
    return User(
            username = username,
            email = email
    )
}